File: typescript/src/client/ReceiptApi.ts
/**
 * Parmana Receipt API.
 *
 * Generate cryptographic execution receipts.
 */

import type {
  Receipt,
} from "../models/index.js";

import type {
  Transport,
} from "../config/Transport.js";

/**
 * Receipt API.
 *
 * Responsibilities
 * ----------------
 * - Generate execution receipts
 *
 * This API does NOT:
 * - execute Business Transactions
 * - replay executions
 * - verify trust records
 * - validate policies
 */
export class ReceiptApi {
  constructor(
    private readonly transport: Transport,
  ) {}

  /**
   * Generate an execution receipt.
   */
  public async generate(
    businessTransactionId: string,
  ): Promise<Receipt> {
    const response =
      await this.transport.send<Receipt>({
        method: "POST",
        path: "/receipt",
        body: {
          businessTransactionId,
        },
      });

    return response.body;
  }
}

File: python/parmana/api/receipt_api.py
"""
Parmana Receipt API.

Generate cryptographic execution receipts.
"""

from __future__ import annotations

from parmana.config.transport import Transport
from parmana.models.receipt import Receipt


class ReceiptApi:
    """
    Receipt API.

    Responsibilities
    ----------------
    - Generate execution receipts
    - Retrieve the latest cached receipt

    This API does NOT:
    - execute Business Transactions
    - replay executions
    - verify trust records
    - validate policies
    """

    def __init__(
        self,
        transport: Transport,
    ) -> None:
        self._transport = transport

    def generate(
        self,
        business_transaction_id: str,
    ) -> Receipt:
        """
        Generate an execution receipt.

        Maps to POST /receipt.

        Parameters
        ----------
        business_transaction_id:
            Business Transaction identifier.

        Returns
        -------
        Receipt.
        """

        return self._transport.send(
            method="POST",
            path="/receipt",
            body={
                "businessTransactionId": business_transaction_id,
            },
            response_model=Receipt,
        )

    def get_latest(
        self,
        business_transaction_id: str,
    ) -> Receipt:
        """
        Retrieve the most recent Receipt without generating a new one.

        Maps to GET /receipt/latest/:id, distinct from `generate()`
        (POST /receipt), which generates a new Receipt.

        Parameters
        ----------
        business_transaction_id:
            Business Transaction identifier.

        Returns
        -------
        Receipt.
        """

        return self._transport.send(
            method="GET",
            path=f"/receipt/latest/{business_transaction_id}",
            response_model=Receipt,
        )
